chain/network/src/peer/peer_actor.rs

```diff
@@ -91,23 +93,20 @@ pub(crate) struct PeerActor {
 
     /// Peer address from connection.
     peer_addr: SocketAddr,
-    /// Peer id and info. Present if outbound or ready.
-    peer_info: DisplayOption<PeerInfo>,
     /// Peer type.
     peer_type: PeerType,
-    /// Peer status.
-    peer_status: PeerStatus,
-    /// Protocol version to communicate with this peer.
-    protocol_version: ProtocolVersion,
+    /// OUTBOUND-ONLY: Handshake specification. For outbound connections it is initialized
+    /// in constructor and then can change as HandshakeFailure and LastEdge messages
+    /// are received. For inbound connections, handshake is stateless.
+    handshake_spec: Option<HandshakeSpec>,
```

Comment:
Not super happy with the amount of unwraps we use for this field. Can we rearange the data to replace `.unwrap`s with `match`es? It seems that perhaps `HandshakeSpec` can be a part of `ConnectionStatus::Outbound`?

Reply:
done